### ISSUE TITLE
TiKV storage and cpu usages are not balance during importing

### ISSUE THREAD

**Issue by cyliu0** [pingcap/tidb-lightning#500]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.
Import 2.4T data to a tidb cluster with 6 tikv nodes
<img width="835" alt="企业微信截图_d4662748-fd2d-482f-af6f-0de45130acec" src="https://user-images.githubusercontent.com/35356271/100721146-ecf0a800-33f9-11eb-9724-f43b272a8ff0.png">

2. What did you expect to see?
TiKV resources are balanced

3. What did you see instead?
TiKV resources are not balanced

4. Versions of the cluster

    - TiDB-Lightning version (run `tidb-lightning -V`):

        ```
        # ./bin/tidb-lightning -V
        Release Version: v4.0.8-27-g8a276c0
        Git Commit Hash: 8a276c0282e59a1783c223b8c6e2bf1990025921
        Git Branch: master
        UTC Build Time: 2020-11-30 06:23:00
        Go Version: go version go1.15.5 linux/amd64
        ```
        
    - TiKV-Importer version (run `tikv-importer -V`)

        ```
        (paste TiKV-Importer version here)
        ```
        
    - TiKV version (run `tikv-server -V`):
       release-4.0-nightly

        ```
        # ./tikv-server -V
        TiKV
        Release Version:   4.0.8
        Edition:           Community
        Git Commit Hash:   4055641d9e51e2e804e07de28cb31454f1394dc9
        Git Commit Branch: HEAD
        UTC Build Time:    2020-11-28 02:18:43
        Rust Version:      rustc 1.42.0-nightly (0de96d37f 2019-12-19)
        Enable Features:   jemalloc mem-profiling portable sse protobuf-codec
        Profile:           dist_release
        ```

    - TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

        ```
        (paste TiDB cluster version here)
        ```

    - Other interesting information (system version, hardware config, etc):

        >
        >

5. Operation logs
   - Please upload `tidb-lightning.log` for TiDB-Lightning if possible
      
[lightning.log](https://github.com/pingcap/tidb-lightning/files/5621683/lightning.log)

   - Please upload `tikv-importer.log` from TiKV-Importer if possible

   - Other interesting logs
   
6. Configuration of the cluster and the task
   - `tidb-lightning.toml` for TiDB-Lightning if possible
   - `tikv-importer.toml` for TiKV-Importer if possible
   - `inventory.ini` if deployed by Ansible

7. Screenshot/exported-PDF of Grafana dashboard or metrics' graph in Prometheus for TiDB-Lightning if possible

**Comment by cyliu0** [comment]
One of the tikv nodes consumed all the space and down. Lightning also failed.
<img width="671" alt="企业微信截图_0e962bbe-8f79-4610-98ea-28a466129b15" src="https://user-images.githubusercontent.com/35356271/101116762-b21c8900-3620-11eb-8044-edc404f718ea.png">

**Comment by glorv** [comment]
From lightning and tikv grafana monitor page, we can see store sizes uneven distribution is caused by region uneven distribution. From tikv summary page, the region count panel clearly showed that at each batch split time,  region count at each store was always not even.

This morning, we reproduced the region's uneven distribution after splitting and scattering regions. With @rleungx 's help, we finally found it was caused by the setting `max-pending-peer-count` default value (16) is too small, so after we 
batch split and scatter several thousand regions, most of the scatter was filter by the pending peer filter, so the new created regions are not actually scattered. 
From pd's following panel, we can see the filter events:
![image](https://user-images.githubusercontent.com/5196885/101606228-18007a80-3a3e-11eb-99a2-865bc3051a68.png)
